src/Worker.Extensions.DurableTask/DurableTaskClientExtensions.cs

```diff
@@ -143,12 +172,31 @@ static string BuildUrl(string url, params string?[] queryValues)
         //       request headers into consideration and generate the base URL accordingly.
         //       More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded.
         //       One potential workaround is to set ASPNETCORE_FORWARDEDHEADERS_ENABLED to true.
-        string baseUrl = request.Url.GetLeftPart(UriPartial.Authority);
+
+        // If HttpRequestData is provided, use its URL; otherwise, get the baseUrl from the DurableTaskClient.
+        // The base URL could be null if:
+        // 1. The DurableTaskClient isn't a FunctionsDurableTaskClient (which would have the baseUrl from bindings)
+        // 2. There's no valid HttpRequestData provided
+        string? baseUrl = ((request != null) ? request.Url.GetLeftPart(UriPartial.Authority) : GetBaseUrl(client))
+            ?? throw new InvalidOperationException("Base URL is null. Either use Functions bindings or provide an HTTP request to create the HttpPayload.");
```

Comment:
this is nitpicky, but I feel this is a bit difficult to read due the combination of ternary expressions the `<condition> ? <true-case> : <false-case>` syntax with the nullable syntax ` <value> ?? <code-to-execute-if-value-is-null>`.

I just feel that there's a lot happening in very little space :) . I would using either the ternary expression, or the nullable syntax, but not both at the same time. The other can be expressed as a simple if-statement.

Not a blocking concern, just a stylistic nit. Feel free to dismiss.

Reply:
Haha I just used this so that I could save more space. I will change this to if-statement to make it more clear. Thanks!